scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala

```diff
@@ -624,6 +626,30 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)])
     new MapSideInput[K, V](o)
   }
 
+  def asSparkeySideInput: SideInput[Map[String, String]] = {
```

Comment:
In that case it's better to have a implicit class that from `SparkeyReader` to `Map` or something Map like. Decoupled from any side input logic and can be used standalone.